packages/block-editor/src/components/inserter/menu.js

```diff
@@ -241,6 +247,23 @@ function InserterMenu(
 			{ showInserterHelpPanel && hoveredItem && (
 				<InserterPreviewPanel item={ hoveredItem } />
 			) }
+			{ showPatternPanel && (
+				<BlockPatternsCategoryDialog
+					rootClientId={ destinationRootClientId }
+					onInsert={ onInsertPattern }
+					category={ selectedPatternCategory }
+					onClose={ () => {
+						if (
+							searchRef.current.ownerDocument.activeElement.classList.contains(
+								'block-editor-inserter__patterns-category'
+							)
```

Comment:
Not really the point of this PR, but the setup with `searchRef`, `focusSearch` and the other places where that string occurs, is hard to decipher.